JWBestPracticeApps/Offline DRM/Offline DRM/ViewController.swift

```diff
@@ -0,0 +1,206 @@
+    /// The local video URL, this gets set from memory if the video is saved and used as the locator for memory videos. Bookmarked data allows us to access the video file in memory since it is stored in a secure location.
+    private var localURL: URL? {
+        let persistedData = UserDefaults.standard.data(forKey: savedDataKey)
+        guard let boomarkData = persistedData else {
+            return nil
+        }
+        var dataStaleStatus = false
+        let url = try? URL(resolvingBookmarkData: boomarkData,
+                           bookmarkDataIsStale: &dataStaleStatus)
+        return dataStaleStatus ? nil : url
+    }
+    /// The player interface, this is from the JWPlayerViewController embedded in a container view.a
+    private var player: JWPlayerProtocol? = nil
+    
+    /// The AVAssetDownloadURLSession to use for managing AVAssetDownloadTasks.
+    private var assetDownloadURLSession: AVAssetDownloadURLSession!
+    /// The delegate queue for the AVAssetDownloadURLSession
+    private let delegateQueue = OperationQueue()
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        /// The keys can be pre-loaded by calling load on the content loader.
+        let url = URL(string: playlistURL)!
+        contentLoader = JWDRMContentLoader(dataSource: keyDataSource, keyManager: keyManager)
+        self.contentLoader?.load(playlist: url)
+        
+        // We construct a playlist either from the remote asset or a local asset if it is saved.
+        var config: JWPlayerConfiguration!
+        
+        if let localURL = localURL {
+            config = self.getPlaylist(local: localURL)
+            // Updates the UI to reflect that the asset is already downloaded.
+            stateLabel?.text = "Saved"
+        } else {
+            config = self.getPlaylist(remote: url)
+        }
+        // We set the content loader to the player.
+        self.player?.contentLoader = self.contentLoader
+        // We configure the player.
+        self.player?.configurePlayer(with: config)
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Create the configuration for the AVAssetDownloadURLSession.
+        let backgroundConfiguration = URLSessionConfiguration.background(withIdentifier: "AAPL-Identifier")
+        assetDownloadURLSession = AVAssetDownloadURLSession(configuration: backgroundConfiguration,
+                                                            assetDownloadDelegate: self, delegateQueue: delegateQueue)
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        guard let playerViewController = segue.destination as? JWPlayerViewController else {
+            return
+        }
+        // Set this controller's player reference to the embedded JWPlayerViewController's.
+        self.player = playerViewController.player
+    }
+    
+    func getPlaylist(remote url: URL) -> JWPlayerConfiguration? {
```

Comment:
Add some header doc to this method so the developers understand about it taking a `URL` and converting it into a useable `JWPlayerConfiguration`, and why you do it this way.